core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -394,4 +396,42 @@ public static Schema schemaFor(Table table, Long snapshotId, Long timestampMilli
 
     return table.schema();
   }
+
+  /**
+   * Fetch the snapshot at the head of the given branch in the given table.
+   *
+   * <p>This method calls {@link Table#currentSnapshot()} instead of using branch API {@link
+   * Table#snapshot(String)} for the main branch to ensure existing tests still goes through the old
```

Comment:
nit/nonblocker: this class seems to be public, not sure if we want to mention the internal testing strategy here. Same goes for `latestSnapshot`

Reply:
good point, I can rephrase this to avoid saying tests